airflow/api_fastapi/views/ui/objects.py

```diff
@@ -0,0 +1,116 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+from __future__ import annotations
+
+from datetime import datetime
+from typing import TYPE_CHECKING
+
+from fastapi import Depends
+from flask import (
+    abort,
+)
+from pendulum.parsing.exceptions import ParserError
+from sqlalchemy import func, select
+from sqlalchemy.orm import Session
+from typing_extensions import Annotated
+
+from airflow.models.dagrun import DagRun, DagRunType
+from airflow.models.taskinstance import TaskInstance
+from airflow.utils.state import DagRunState, TaskInstanceState
+
+if TYPE_CHECKING:
+    from sqlalchemy.orm import Session
+from airflow.api_fastapi.db.common import get_session
+from airflow.api_fastapi.views.router import AirflowRouter
+from airflow.utils import timezone
+
+object_router = AirflowRouter(tags=["Object"])
+
+
+@object_router.get("/object/historical_metrics_data", include_in_schema=False)
```

Comment:
I didn't like the `/object/` nomenclature in `views.py`. That was merely an attempt to mark which endpoints returned objects instead of FAB views.

Let's take the opportunity to give a more descriptive name. In this case, we are using the historical metrics for a dashboard view. So let's make it a `dashboard_router`

Reply:
Good idea. The final route could be `ui/dashboard/historical_metrics_data`

Reply:
Make sense! I made the changes that were needed. Thanks for the both! 

Indeed, we should take this as an advantage to shape them properly. Please make all these comments because I am first doing lift and shift to see if it would be enough. Please also include if you need any additional things from the UI end.